e2e-tests/src/test/java/uk/nhs/adaptors/gp2gp/e2e/EhrExtractTest.java

```diff
@@ -217,10 +231,45 @@ private void assertHappyPathWithDocs(String conversationId, String fromODSCode,
         var ehrContinue = (Document) waitFor(() -> Mongo.findEhrExtractStatus(conversationId).get(EHR_CONTINUE));
         assertThatExtractContinueMessageWasSent(ehrContinue);
 
-        waitFor(() -> assertThat(assertThatExtractCommonMessageWasSent(conversationId)).isTrue());
-
         var ackPending = (Document) waitFor(() -> Mongo.findEhrExtractStatus(conversationId).get(ACK_TO_PENDING));
         assertThatAcknowledgementPending(ackPending, ACCEPTED_ACKNOWLEDGEMENT_TYPE_CODE);
+
+        var sentToMhs = (Document) waitFor(() -> fetchSentToMhsForDocuments(conversationId));
+        assertThat(sentToMhs.get("messageId")).isNotNull();
+        assertThat(sentToMhs.get("sentAt")).isNotNull();
+        assertThat(sentToMhs.get("taskId")).isNotNull();
+    }
+
+    private void assertEhrExtractSentAsAttachment(String conversationId) {
+        var gpcAccessStructured = waitFor(() -> accessStructuredWithAttachmentThatHasBeenSent(conversationId));
+        assertThat(gpcAccessStructured.get("documentId")).isNotNull();
+        assertThat(gpcAccessStructured.get("objectName")).isNotNull();
+        assertThat(gpcAccessStructured.get("accessedAt")).isNotNull();
+        assertThat(gpcAccessStructured.get("taskId")).isNotNull();
+        assertThat(gpcAccessStructured.get("messageId")).isNotNull();
+        assertThat(gpcAccessStructured.get("sentToMhs")).isNotNull();
+    }
+
+    private Document fetchSentToMhsForDocuments(String conversationId) {
+        var gpcAccessDocument = (Document) Mongo.findEhrExtractStatus(conversationId).get(GPC_ACCESS_DOCUMENT);
+        var documentList = gpcAccessDocument.get("documents", Collections.emptyList());
+        if (documentList == null || documentList.isEmpty())
+            return null;
```

Comment:
this null will throw null pointer exception in the test right? e.g. assertThat(sentToMhs.get("messageId")).isNotNull(); would throw exception? if that is the case maybe null is some king of exception that would have to be thrown?